gr-blocks/lib/file_source_impl.cc

```diff
@@ -123,20 +129,40 @@ namespace gr {
         throw std::runtime_error("can't open file");
       }
 
-      //Check to ensure the file will be consumed according to item size
-      GR_FSEEK(d_new_fp, 0, SEEK_END);
-      uint64_t file_size = GR_FTELL(d_new_fp);
+      struct stat st;
 
-      // Make sure there will be at least one item available
-      if ((file_size / d_itemsize) < (start_offset_items+1)) {
-        if (start_offset_items) {
-          GR_LOG_WARN(d_logger, "file is too small for start offset");
-        }
-        else {
-          GR_LOG_WARN(d_logger, "file is too small");
+      if(fstat(fileno(d_new_fp), &st)) {
+        GR_LOG_ERROR(d_logger, boost::format("%s: %s") % filename % strerror(errno));
+        throw std::runtime_error("can't fstat file");
+      }
+      if(S_ISREG(st.st_mode)) {
```

Comment:
Oh, this is for the automatic determination if it's seekable?

Reply:
Yes, but I think it needs to be fixed for Windows (functions need leading underbars).